torchao/quantization/prototype/qat/api.py

```diff
@@ -4,11 +4,75 @@
 # This source code is licensed under the license found in the
 # LICENSE file in the root directory of this source tree.
 
-from typing import Any, List
+from dataclasses import dataclass
+from enum import Enum
+from typing import Any, List, Optional
 
 import torch
 
 from torchao.quantization.unified import TwoStepQuantizer
+from torchao.quantization.quant_primitives import ZeroPointDomain
+
+
+class QuantizationGranularity(Enum):
+    PER_CHANNEL = "per_channel"
+    PER_TOKEN = "per_token"
+    PER_GROUP = "per_group"
+
+
+@dataclass
+class FakeQuantizeConfig:
+    """
+    Config for how to fake quantize weights or activations.
+
+    args:
+        bit_width: number of bits to simulate during fake quantization
+        granularity: granularity of scales and zero points, one of:
+            'per_token', 'per_channel', or 'per_group'
+        group_size: size of each group for 'per_group' granularity
+        symmetric: whether to use symmetric (default) or asymmetric quantization
```

Comment:
is it better to do `is_symmetric`, or maybe we should use `MappingType` I feel, we now have 3 types: https://github.com/pytorch/ao/blob/dec03131484b5717e9fb8b6c0c25f41ec30e9fef/torchao/quantization/quant_primitives.py#L36